ckanext/saml2/plugin.py

```diff
@@ -663,21 +713,17 @@ def staff_login(self):
         return self.login()
 
     def edit(self, id=None, data=None, errors=None, error_summary=None):
-        c = p.toolkit.c
-        c.allow_user_changes = p.toolkit.asbool(
-            config.get('ckan.saml2.allow_user_changes', False))
-        c.is_allow_update = saml2_get_user_info(c.user).first().allow_update
+        saml2_check_for_user_update(id)
         return super(Saml2Controller, self).edit(id, data, errors, error_summary)
 
     def _save_edit(self, id, context):
-        user_custom_profile_data = p.toolkit.request.params.get(
-                                        'user_custom_profile_data')
-        name = p.toolkit.request.params.get('name')
-        user_info_query = saml2_get_user_info(id)
-        if id != name:
-            user_info_query.update({'user_name': name})
-        if user_custom_profile_data is not None:
-            user_info_query.update({'allow_update': True})
-        else:
-            user_info_query.update({'allow_update': False})
+        name_id = saml2_get_user_name_id(id)
+        if name_id is not None:
+            c = p.toolkit.c
+            user_custom_profile_data = p.toolkit.request.params.get(
+                                            'user_custom_profile_data')
+            if user_custom_profile_data is not None:
+                c.allow_update_checkbox = True
+            else:
+                c.allow_update_checkbox = False
```

Comment:
Sorry, i do not understand purpose of this check.
PS is there any difference between `user_custom_profile_data` ans Saml2Info.allow_update?